mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py

```diff
@@ -48,3 +55,82 @@ class WikiClusteringP2P(AbsTaskClustering, MultilingualTask):
         n_samples={"test": 71680},
         avg_character_length={"test": 625.3},
     )
+
+
+class WikiClusteringFastP2P(AbsTaskClusteringFast, MultilingualTask):
+    metadata = TaskMetadata(
+        name="WikiClusteringFastP2P",
+        description="Clustering of wikipedia articles inspired by BlubrbsClusteringP2P. Labels are taken from top-level categories of the respective languages (e.g., https://lv.wikipedia.org/wiki/Kategorija:Pamatkategorijas).",
+        reference="https://github.com/Rysias/wiki-clustering",
+        dataset={
+            "path": "ryzzlestrizzle/multi-wiki-clustering-p2p",
+            "revision": "d4d92f8f28be71035be6a96bdfd4e200cf62faa8",
+        },
+        type="Clustering",
+        category="p2p",
+        eval_splits=["test"],
+        eval_langs=_LANGUAGES,
+        main_score="v_measure",
+        date=("2001-01-15", "2024-04-15"),
+        form=["written"],
+        domains=["Encyclopaedic"],
+        task_subtypes=["Thematic clustering"],
+        license="cc-by-sa-3.0",
+        socioeconomic_status="mixed",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="created",
+        bibtex_citation="",  # None exists
+        n_samples={"test": 2048},
+        avg_character_length={"test": 625.3},
+    )
+
+    def dataset_transform(self):
+        ds = dict()
+        for lang in self.hf_subsets:
+            labels = []
+            sentences = []
+            ds[lang] = dict()
+            lang_dict = dict()
+            for split in self.metadata.eval_splits:
+                labels.extend(
+                    list(
+                        itertools.chain.from_iterable(
+                            self.dataset[lang][split]["labels"]
+                        )
+                    )
+                )
+                sentences.extend(
+                    list(
+                        itertools.chain.from_iterable(
+                            self.dataset[lang][split]["sentences"]
+                        )
+                    )
+                )
+
+                # Remove sentences and labels with only 1 label example.
+                unique_labels, counts = np.unique(labels, return_counts=True)
+                solo_label_idx = np.where(counts == 1)
+                solo_labels = unique_labels[solo_label_idx]
+                for solo_label in solo_labels:
+                    loc = labels.index(solo_label)
```

Comment:
This is quite a slow solution considering that you will have to do a linear search for every lone label.
You could perhaps use `np.isin`:
```python
is_solo = np.isin(labels, solo_labels)
split_ds = Dataset.from_dict({"labels": labels, "sentences": sentences})
split_ds = split_ds.select(np.nonzero(is_solo))
lang_dict.update({split: split_ds})
```

Reply:
hmm I wonder if it's related to [this](https://numpy.org/doc/stable/reference/generated/numpy.isin.html):
```
Returns:
isin ndarray, bool
``

Reply:
Will push a fix for this as well. Thanks for the suggestions!